compiler/x/codegen/SIMDTreeEvaluator.cpp

```diff
@@ -0,0 +1,237 @@
+      }
+   TR_ASSERT(opCode != BADIA32Op, "SIMD Store Evaluator: unsupported fill size");
+
+   TR::Instruction* instr = generateMemRegInstruction(opCode, node, tempMR, valueReg, cg);
+
+   cg->decReferenceCount(valueNode);
+   tempMR->decNodeReferenceCounts(cg);
+   if (node->getOpCode().isIndirect())
+      cg->setImplicitExceptionPoint(instr);
+   return NULL;
+   }
+
+TR::Register* OMR::X86::TreeEvaluator::SIMDsplatsEvaluator(TR::Node* node, TR::CodeGenerator* cg)
+   {
+   TR::Node* childNode = node->getChild(0);
+   TR::Register* childReg = cg->evaluate(childNode);
+
+   uint8_t shufconst = 0;
+   switch (node->getDataType())
+      {
+      case TR::VectorInt32:
+      case TR::VectorFloat:
+         shufconst = 0;
+         break;
+      case TR::VectorInt64:
+      case TR::VectorDouble:
+         shufconst = 0x44;
+         break;
+      default:
+         TR_ASSERT(false, "SIMD Splats Evaluator: unsupported data type");
+         break;
+      }
+
+   TR::Register* resultReg = cg->allocateRegister(TR_VRF);
+   generateRegRegImmInstruction(PSHUFDRegRegImm1, node, resultReg, childReg, shufconst, cg);
+
+   node->setRegister(resultReg);
+   cg->decReferenceCount(childNode);
+   return resultReg;
+   }
+
+enum BinaryArithmetic_t : uint32_t
+   {
+   BinaryArithmeticInvalid,
+   BinaryArithmeticAdd,
+   BinaryArithmeticSub,
+   BinaryArithmeticMul,
+   BinaryArithmeticDiv,
+   BinaryArithmeticNum
+   };
+static const TR_X86OpCodes BinaryArithmeticOpCodes[TR::NumOMRTypes][BinaryArithmeticNum] =
+   {
+   //  Invalid,       Add,         Sub,       Mul,         Div
+   { BADIA32Op, BADIA32Op,   BADIA32Op, BADIA32Op,   BADIA32Op }, // NoType
+   { BADIA32Op, BADIA32Op,   BADIA32Op, BADIA32Op,   BADIA32Op }, // Int8
+   { BADIA32Op, BADIA32Op,   BADIA32Op, BADIA32Op,   BADIA32Op }, // Int16
+   { BADIA32Op, BADIA32Op,   BADIA32Op, BADIA32Op,   BADIA32Op }, // Int32
+   { BADIA32Op, BADIA32Op,   BADIA32Op, BADIA32Op,   BADIA32Op }, // Int64
+   { BADIA32Op, BADIA32Op,   BADIA32Op, BADIA32Op,   BADIA32Op }, // Float
+   { BADIA32Op, BADIA32Op,   BADIA32Op, BADIA32Op,   BADIA32Op }, // Double
+   { BADIA32Op, BADIA32Op,   BADIA32Op, BADIA32Op,   BADIA32Op }, // Address
+   { BADIA32Op, BADIA32Op,   BADIA32Op, BADIA32Op,   BADIA32Op }, // VectorInt8
+   { BADIA32Op, BADIA32Op,   BADIA32Op, BADIA32Op,   BADIA32Op }, // VectorInt16
+   { BADIA32Op, BADIA32Op,   BADIA32Op, BADIA32Op,   BADIA32Op }, // VectorInt32
+   { BADIA32Op, BADIA32Op,   BADIA32Op, BADIA32Op,   BADIA32Op }, // VectorInt64
+   { BADIA32Op, BADIA32Op,   BADIA32Op, BADIA32Op,   BADIA32Op }, // VectorFloat
+   { BADIA32Op, ADDPDRegReg, BADIA32Op, MULPDRegReg, BADIA32Op }, // VectorDouble
+   { BADIA32Op, BADIA32Op,   BADIA32Op, BADIA32Op,   BADIA32Op }, // Aggregate
+   };
+
+TR::Register* OMR::X86::TreeEvaluator::BinaryArithmeticEvaluator(TR::Node* node, TR::CodeGenerator* cg)
+   {
+   BinaryArithmetic_t arithmetic = BinaryArithmeticInvalid;
+   switch (node->getOpCodeValue())
```

Comment:
Yes, as its name indicates, it is a generic evaluator for binary arithmetic. Comment has been added.